airflow/providers/asana/operators/asana_tasks.py

```diff
@@ -0,0 +1,188 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Dict, Optional
+
+from airflow.models import BaseOperator
+from airflow.providers.asana.hooks.asana import AsanaHook
+from airflow.utils.decorators import apply_defaults
+
+
+class AsanaCreateTaskOperator(BaseOperator):
+    """
+    This operator can be used to create Asana tasks. For more information on
+    Asana optional task parameters, see https://developers.asana.com/docs/create-a-task
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:AsanaCreateTaskOperator`
+
+    :param asana_conn_id: The Asana connection to use.
+    :type asana_conn_id: str
+    :param name: Name of the task.
+    :type name: str
+    :param task_parameters: Any of the optional task creation parameters.
+        See https://developers.asana.com/docs/create-a-task for a complete list.
+        You must specify at least one of 'workspace', 'parent', or 'projects'.
```

Comment:
Is it possible to leverage default values from the connection object?
should be possible to create custom connection that contains these and read the default from it (allowing the user to overwrite if needed in the operator itself). what do you think?

Reply:
Good idea -- will do!

Reply:
So I gave this a try (in the hook) -- curious if this is along the lines of what you had in mind, any further thoughts much appreciated.

* I think the defaults were only likely to be helpful in create and find (do you disagree?), so those were the only places I used them.
* create expects a "projects" parameter but find expects "project". I let the user specify either in the extras and then (where possible) just renamed the key as needed - possibly overkill (maybe it would be better to just require the user to call the parameter "projects") but I was worried this would be more confusing otherwise.

Reply:
@jmelot I was referring to something like: https://github.com/apache/airflow/pull/14724
Where you can define workspace, project etc... in the connection itself these will be the defaults and you can overwrite the defaults in the operator if you want to.

Reply:
Ahh ok got it, that makes sense and would be easier for people to use than what I did. I will give it a try. Thanks!

Reply:
@RosterIn I gave this a try. Workspace and project were the only parameters that jumped out at me as meriting their own fields in the form, but please let me know if you think something else would be helpful as well. Users can override these fields by passing them into the operator parameters. 

I think this was a great suggestion, the connection form should be much clearer now - thanks again!

Reply:
sounds reasonable.